### ISSUE TITLE
New higher geography

### ISSUE THREAD

**Issue by diannakrejsa** [ArctosDB/data-migration#149]
Can we have "Africa, Zimbabwe, Mashonaland East Province" added to possible higher geography code tables? 

From Wikipedia:
Mashonaland is a region in northern Zimbabwe. It is the native place of the Shona people.
Currently, Mashonaland is divided into three provinces,
    Mashonaland West
    Mashonaland Central
    Mashonaland East
![image](https://user-images.githubusercontent.com/36166607/68685413-b8305b00-052f-11ea-8ec6-4a31931ca074.png)

**Comment by Jegelewicz** [comment]
Added. Next time just give me the wikipedia link!

**Comment by diannakrejsa** [comment]
Sounds good, I'll do that!

**Comment by diannakrejsa** [comment]
Curious if the higher geography "South America, Ecuador, Francisco de Orellana" should be updated to "South America, Ecuador, Orellana". It appears that the province is named after the explorer Francisco de Orellana but isn't formally called that?

https://en.wikipedia.org/wiki/Orellana_Province

**Comment by Jegelewicz** [comment]
That is correct and "Francisco de Orellana" is a canton of "Orellana" Province. @dustymc can you fix this one for us?

**Comment by dustymc** [comment]
you should be able to update it, or I can as well

**Comment by diannakrejsa** [comment]
May I have the higher geography: "Asia, Indonesia, Central Sulawesi, Sigi Regency" added to Arctos?
https://en.wikipedia.org/wiki/Sigi_Regency

**Comment by Jegelewicz** [comment]
Created as

Asia, Indonesia, Central Sulawesi, Sigi Regency, Indonesian Archipelago, Sulawesi

**Comment by diannakrejsa** [comment]
Great, thank you.

**Comment by diannakrejsa** [comment]
Can I request a new higher geography? It's a municipality within Oaxaca, Mexico.
https://en.wikipedia.org/wiki/San_Francisco_del_Mar
North America, Mexico, Oaxaca, San Francisco del Mar

**Comment by diannakrejsa** [comment]
Another couple in Mexico if possible:

North America, Mexico, Michoacan, Nueva Italia
https://en.wikipedia.org/wiki/Nueva_Italia,_Michoac%C3%A1n

North America, Mexico, Oaxaca, Santiago Pinotepa Nacional or North America, Mexico, Oaxaca, Pinotepa Nacional 
https://en.wikipedia.org/wiki/Pinotepa_Nacional

**Comment by Jegelewicz** [comment]
> North America, Mexico, Oaxaca, San Francisco del Mar

It is part of the Juchitán District in the west of the Istmo de Tehuantepec region.
Use North America, Mexico, Oaxaca, Juchitan District and put San Francisco del Mar in specific locality

**Comment by Jegelewicz** [comment]
> North America, Mexico, Oaxaca, Santiago Pinotepa Nacional or North America, Mexico, Oaxaca, Pinotepa Nacional

It is located in the Jamiltepec District.
Use North America, Mexico, Oaxaca, Jamiltepec District and put Santiago Pinotepa Nacional in specific locality.

**Comment by Jegelewicz** [comment]
> North America, Mexico, Michoacan, Nueva Italia

Nueva Italia is a city - cities belong in specific locality.
Use North America, Mexico, Michoacan, Mugica

**Comment by diannakrejsa** [comment]
Does this one have a different district/name we should file it under?

North America, Mexico, Michoacan, Nueva Italia
https://en.wikipedia.org/wiki/Nueva_Italia,_Michoac%C3%A1n

**Comment by Jegelewicz** [comment]
It is a city in 

North America, Mexico, Michoacan, Mugica

**Comment by diannakrejsa** [comment]
Thank you!

**Comment by diannakrejsa** [comment]
Couple more, from China:

1. Perhaps "Asia, China, Yunnan" (currently in Arctos) should be "Asia China, Yunnan Province" (this aligns with how other provinces within China are written on the db, and I think it's a proper name for the locality?

2. My data say, simply, "Xishuangbanna, Yunnan, China" on the tag, so that would be this string to add as a higher geography?: Asia, China, Yunnan Province, Xishuangbanna Dai Autonomous Prefecture

2.2 My data say, simply, "Menglun, Yunnan, China" on the tag and I believe Menglun is a town within Xishuangbanna Prefecture, but it is also within the County of Mengla. Would you think that is the same higher geography as above, or provide a level of specificity further?: 
Asia, China, Yunnan Province, Xishuangbanna Dai Autonomous Prefecture [SPEC_LOCALITY: Menglun]
or
Asia, China, Yunnan Province, Xishuangbanna Dai Autonomous Prefecture, Mengla County [SPEC_LOCALITY: Menglun]

Some links for really going down the rabbit hole:
https://en.wikipedia.org/wiki/Mengla_County
https://en.wikipedia.org/wiki/Xishuangbanna_Dai_Autonomous_Prefecture

**Comment by diannakrejsa** [comment]
Any thoughts on these Chinese localities?

**Comment by Jegelewicz** [comment]
@diannakrejsa sorry - I was out yesterday. Let me take a look.

**Comment by Jegelewicz** [comment]
> Perhaps "Asia, China, Yunnan" (currently in Arctos) should be "Asia China, Yunnan Province" (this aligns with how other provinces within China are written on the db, and I think it's a proper name for the locality?

Changed Yunnan to Yunnan Province and Asia, China, Yunnan, Lijiang to 
Asia, China, Yunnan Province, Lijiang

**Comment by Jegelewicz** [comment]
> My data say, simply, "Xishuangbanna, Yunnan, China" on the tag, so that would be this string to add as a higher geography?: Asia, China, Yunnan Province, Xishuangbanna Dai Autonomous Prefecture

Added Asia, China, Yunnan Province, Xishuangbanna Dai Autonomous Prefecture

**Comment by diannakrejsa** [comment]
Thank you!

**Comment by Jegelewicz** [comment]
> My data say, simply, "Menglun, Yunnan, China" on the tag and I believe Menglun is a town within Xishuangbanna Prefecture, but it is also within the County of Mengla. Would you think that is the same higher geography as above, or provide a level of specificity further?

I would use "Xishuangbanna Dai Autonomous Prefecture" and put Mengla County in specific locality. 

China looks like a bit of a mess - three of the Yunnan geographies were spelled "Yunaan"! I corrected those, but the higher geography in China really has 4 levels instead of three and all of their "counties" are prefectures or "cities" (meaning city prefectures). I'm going to put this in the main repository for consideration when someone working in China has time to take a good look at it.

**Comment by diannakrejsa** [comment]
Can you add a new higher geography for the state of Querétaro, Mexico? I don't think it exists yet, and I do think it's part of independent state (not nested within something else. What do you think?
https://en.wikipedia.org/wiki/Quer%C3%A9taro

**Comment by dustymc** [comment]
http://arctos.database.museum/geography.cfm?STATE_PROV=Queretaro

http://handbook.arctosdb.org/documentation/higher-geography.html#guidelines-for-geographic-terms-in-arctos

**Comment by diannakrejsa** [comment]
Oh, weird. I searched for " Queretaro" in Any Geog of the Find Higher Geography query tool and it yielded no results. I also searched for "Mexico" to try to view a full list. I wonder what I was doing wrong, I'll read on for more info in the handbook..

**Comment by Jegelewicz** [comment]
> I searched for " Queretaro" in Any Geog of the Find Higher Geography query tool and it yielded no results.

I never use that field - it is unpredictable at best.

> I also searched for "Mexico" to try to view a full list. 

The results only display the first 100 or something like that and "Q" was probably beyond that threshold. This has been a problem for me in the past an we should probably look at changing that.  I suggest that you also try searching the term in the "state" and "county" fields using a % after it just in case it somehow also includes "prefecture" or some other term.

**Comment by sharpphyl** [comment]
Recommended search that returns results for North America, Mexico, Queretaro

![Screen Shot 2020-03-06 at 1 52 50 PM](https://user-images.githubusercontent.com/15368365/76121828-d3791780-5fb1-11ea-901d-77555d16b11e.png)

**Comment by diannakrejsa** [comment]
Great suggestions! Thank you

**Comment by diannakrejsa** [comment]
I'm sure there is a tool to do this, and perhaps I am missing the how-to on it in the handbook, but:

Is there a way to find which existing Arctos records need to be georeferenced?

**Comment by dustymc** [comment]
<img width="440" alt="Screen Shot 2020-03-17 at 10 57 00 AM" src="https://user-images.githubusercontent.com/5720791/76886455-0fc43780-683e-11ea-972b-e2d81afcc895.png">

**Comment by Jegelewicz** [comment]
@diannakrejsa I was thinking last night that it would be awesome if you could write up the process you are using to have your students georeference - all the way from how you found stuff that needed it. It could help so many people!

**Comment by diannakrejsa** [comment]
Sure! I've already made some instructional material and a few videos for the students working on this. I can compress it into a single document (not sure anyone wants to deal with my videos anyway, lol) that is angled toward database managers.

**Comment by Jegelewicz** [comment]
Videos might be awesome! They can be posted on the Arctos YouTube channel....

**Comment by campmlc** [comment]
Videos would be great!!

**Comment by diannakrejsa** [comment]
So, before I finalize the instructions for "Georeferencing in Arctos - for data managers" I have a question: I successfully loaded a handful of now georeferenced records, thirteen that shared a COLLECTING_EVENT_ID (11355047). Will Arctos eventually eliminate these records as being "Not Georeferenced"? Currently they all still show up in the search for records left to be georeferenced, even though they have lat/longs; they have two locality events/boxes when examining the record. I don't mind the two boxes, but I would hope they would no longer show up in the list to georeference. Will that happen eventually?
<img width="1038" alt="Screen Shot 2020-03-29 at 5 07 19 PM" src="https://user-images.githubusercontent.com/36166607/77862270-03e25900-71e0-11ea-85b9-3aef5b847baa.png">

**Comment by mkoo** [comment]
No, you should manually merge those events or localities.
We cover it in our MVZ Arctos Tutorial under merging localities
<https://docs.google.com/document/d/1dsUTONLnj6IFJCVE8PntOoWkuqJNFxDrZxteLy-rIyc/edit#heading=h.mqbclvl8xxie>
which is also linked in the Arctos Handbook.

**Comment by diannakrejsa** [comment]
I don't have "Location" as an option under the Manage Data tab. Can someone add that access to my account at some point? Thank you.

**Comment by campmlc** [comment]
done

**Comment by diannakrejsa** [comment]
Awesome, seems the merge was successful. I let it sync a bit and now those records do not appear as part of the list of "non georeferenced". Thanks everyone

**Comment by dustymc** [comment]
isGeoreferenced=false is looking for "has an ungeoreferenced locality" NOT "has at least one georeferenced locality." That probably needs an Issue in the main repo if we need adjustments.

I don't think cache would have been a factor in this.

Arctos will automatically merge duplicate localities - http://handbook.arctosdb.org/documentation/locality.html#maintenance

Keeping the "old" locality stack, in conjunction with locality archives, provides a full edit history. How you use that is a curatorial call, but the capability exists.

Note the map border and the 'map key/tools' link on specimendetail - eg https://arctos.database.museum/guid/ASNHC:Mamm:10 has two georeferenced localities, and neither of them have the centroid inside the asserted geography.

<img width="914" alt="Screen Shot 2020-03-30 at 7 24 07 AM" src="https://user-images.githubusercontent.com/5720791/77923666-9c67f000-7257-11ea-9b14-a6342a70e15b.png">

**Comment by diannakrejsa** [comment]
Ah, this Rocker b Ranch is quite large and sits on two counties. It's probably supposed to have higher geography in Irion County.

**Comment by diannakrejsa** [comment]
[Georeferencing on Arctos - for data managers.docx](https://github.com/ArctosDB/data-migration/files/4403615/Georeferencing.on.Arctos.-.for.data.managers.docx)

Here is a document with the steps taken--from finding the records to georeferencing to final bulkload. Please edit to refine best practices if you have better suggestions on each step, especially concerning the generation of centroids for things like states and countries. That method was something I was taught to do but I'm not sure it is the most precise method. 

I think switching the language for the isGeoreferenced=false tool would be good.

**Comment by dustymc** [comment]
We should better document the built-in tools as well. Arctos automagically georeferences everything that it can, so you can eg

http://arctos.database.museum/guid/ASNHC:Mamm:1
<img width="61" alt="Screen Shot 2020-03-30 at 8 10 52 AM" src="https://user-images.githubusercontent.com/5720791/77928995-fff51c00-725d-11ea-94b6-eca6dd347fc6.png">

<img width="82" alt="Screen Shot 2020-03-30 at 8 11 09 AM" src="https://user-images.githubusercontent.com/5720791/77929030-097e8400-725e-11ea-8d7c-e97dda519b7a.png">

<img width="378" alt="Screen Shot 2020-03-30 at 8 11 35 AM" src="https://user-images.githubusercontent.com/5720791/77929074-1a2efa00-725e-11ea-9c78-6744225650eb.png">

and save to do MOST of that. The big gap is that doesn't calculate error, but...

<img width="147" alt="Screen Shot 2020-03-30 at 8 12 24 AM" src="https://user-images.githubusercontent.com/5720791/77929223-434f8a80-725e-11ea-95dd-f325b36877ab.png">

<img width="150" alt="Screen Shot 2020-03-30 at 8 13 05 AM" src="https://user-images.githubusercontent.com/5720791/77929271-4fd3e300-725e-11ea-899c-81f443a099bd.png">

does - 2 clicks gets decent data SOME of the time. (Geolocate can be a little twitchy - http://handbook.arctosdb.org/documentation/locality.html#specific-locality is important!)

Even when that isn't as smooth as it could be, I think it's impossible to overstate the impact of using spatial tools to produce spatial data - people just don't tend to save things that say "Texas" and map to Nebraska, where that's really common when there's not a map involved. If this was much slower than anything involving a spreadsheet - and I don't think it is - I'd still want to something that does involve maps and does not involve copy-paste.

I could also make the precalculated coordinates available to you, if that somehow works better in your workflow. Or they're in locality if you're writing SQL:


 s$elevation           | bigint                      |           |          | 
 s$geography           | character varying(4000)     |           |          | 
 s$dec_lat             | double precision            |           |          | 
 s$dec_long            | double precision            |           |

**Comment by diannakrejsa** [comment]
I'm probably missing some of what you're describing (sorry! i'm not an Arctos super-user by any means yet), but the main reason why I went about georeferencing outside of Arctos is being able to have students work on it without assigning them all the locality editing controls. Teresa and I talked about it briefly last Thursday, but I'm open to whatever is the preferred method for having students work on georeferencing (especially remotely where they can't ask me questions directly) and I can have some power as a stop gap to review the spatial data they're generating before it goes online. 

Right now they're working on data that hasn't been uploaded to Arctos yet, so they're currently working in flat files. My steps are thereby biased toward working on it in flat files and doing the Specimen Event bulkloader later for the records that are already in Arctos and need to be georeferenced. Suggestions for best practices in this situation?

**Comment by dustymc** [comment]
No suggestions, that all makes perfect sense. Let me know if I can help somehow - make Arctos better, SQL something in or out, whatever.

**Comment by Jegelewicz** [comment]
I agree - my only concern is that localities get merged that shouldn't be. I really don't like the merge thingee without some further review and checking to make sure you aren't modifying a locality that someone else doesn't want changed. It might be weeks or months before such a merge is figured out and undoing those things isn't easy.
